File: 2427-first-letter-to-appear-twice/2427-first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
        map<char, int> table;

        for (auto ch: s) {
            table[ch]++;
            if (table[ch] == 2) {
                return ch;
            }
        }

        return 's';
    }
};

File: 2427-first-letter-to-appear-twice/2427-first-letter-to-appear-twice.java
class Solution {
    public char repeatedCharacter(String s) {
        int[] chars = new int[26];

        for (char ch: s.toCharArray()) {
            chars[ch - 'a']++;
            if (chars[ch - 'a'] == 2) {
                return ch;
            }
        }

        return '0';
    }
}